ci/miri.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+set -e
+
+MIRI_NIGHTLY=nightly-$(curl -s https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/miri)
+echo "Installing latest nightly with Miri: $MIRI_NIGHTLY"
+rustup set profile minimal
+rustup default "$MIRI_NIGHTLY"
+
+rustup component add miri
+cargo miri setup
+
+# FIXME: investigate memory leaks
+cargo miri test --all-features -- -Zmiri-ignore-leaks
```

Comment:
I didn't do any further investigation for why Miri sees a memory leak.
This might or might not be https://github.com/rust-lang/miri/issues/940.

Reply:
One of the leaked allocations is allocated here:
```
note: inside call to `std::string::String::push_str` at src/lean.rs:87:13
   --> src/lean.rs:87:13
    |
87  |             owned.push_str("Hello?.. ");
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside call to `lean::tests::stress_test_owned` at src/lean.rs:75:5
   --> src/lean.rs:75:5
    |
75  | /     fn stress_test_owned() {
76  | |         let mut expected = String::from("Hello... ");
77  | |         let mut cow: Cow<str> = Cow::borrowed("Hello... ");
78  | |
...   |
92  | |         assert_eq!(expected, cow.into_owned());
93  | |     }
    | |_____^
```
Doesn't look like there is anything being put in a `static` here, so this is likely a genuine leak?